test-harness/modules/mocker/endpoints/marathon.py

```diff
@@ -389,6 +392,19 @@ def enable_nginx_task(self, *_):
         with self._context.lock:
             self._context.data["endpoint-content"]["apps"].append(NGINX_TASK_ENABLED)
 
+    def apps_response_raw(self, content):
```

Comment:
`Clean code` preaches that code should tell a story. Let's keep it in mind and analyze these two functions.

`apps_response_raw` at first glance seems like a getter to me - "this function returns apps_response in raw format", but only after reading the docstring and  the code, one realizes that this is not a case - this function instructs the Marathon mock treat stored reply as binary blob/do not try to convert it to JSON string. The same thing goes for `app_response` function - again, sound like a getter, but in fact this method is setting Marathon endpoint data to certain value.

What do we want o achieve here? The story in my opinion goes  "please set the Mock's `/v2/apps` response to this-and-that. In certain cases I need you to treat the input data litteraly and do not try to parse it as JSON".

So, for me an ideal solution would be a function `set_apps_response` with two parameters:
* response to set
* how the response should be treated (`binary` vs. `data`), by default `data`

We not only get a better readability but also the type of the data becomes inherent to the data itself, it's not possible to change after it was set. This makes it much harder to people writing tests to make mistakes and easier to understand the code for people reading tests.

The aux_data can parameter that is passed to the function can be anything really - we can make it a dict with these to params as keys and the input treated as `data` if the `type` key is not present.

What do you think?